jobs/JREGIONAL_GET_EXTRN_MDL_FILES

```diff
@@ -57,12 +57,13 @@
 #-----------------------------------------------------------------------
 #
 . $USHdir/source_util_funcs.sh
+
 sections=(
   user
   nco
   workflow
-  task_get_extrn_lbcs
-  task_get_extrn_ics
+  task_get_extrn_lbcs.envars
+  task_get_extrn_ics.envars
```

Comment:
Several sections in the `ush/config_defaults.yaml` file got new `execution` and `envars` blocks. In the bash script, we "source" the YAML using a conversion from YAML-> bash in which only depth-2 configs are supported. This now allows us to only source the section containing the variables needed in the bash script.

Reply:
I googled `"envars" or "envvars"` and it seems the software community prefers the latter. I also `grep`ed the `uwtools` codebase and found that `envar` is a term used in a Rocoto context, while `envvar` is the preferred `uwtools` term (which agrees with e.g. our use of `envcmd` for environment _commands_). I think Rocoto is on the wrong side of this name choice, but if this bit of code is meant to align with Rocoto mainly, maybe it's appropriate. I thought I'd mention the disagreement, though, in case this wasn't an intentional choice.

FWIW, google's AI is adamant about it:
<img width="622" alt="image" src="https://github.com/user-attachments/assets/b13bcca0-08eb-4042-b252-a288d0974ed7">

Reply:
Reading further, it does look like this is all about (no surprise given the topic of this PR) the Rocoto context, so I think it's the right choice, which you already knew.

Reply:
Actually, I think this part is my Rocoto experience/bias showing. This particular piece is about separating the bash environment variables to be sourced from the Rocoto resources bit, which fall under the `execution` sections at the same level in the config. 

I will make the change here to `envvars` since this is a piece of the code _not_ meant to be coupled tightly to Rocoto.